app/promoCodeFirstRunStorage.js

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const electron = require('electron')
+const path = require('path')
+const fs = require('fs')
+
+const promoCodeFileRelativeUserDataPath = 'promoCode'
+
+function getPromoCodeFileAbsolutePath () {
+  return path.join(electron.app.getPath('userData'), promoCodeFileRelativeUserDataPath)
+}
+
+module.exports = {
+  writeFirstRunPromoCodeSync: promoCode => {
+    const promoCodeOutputPath = getPromoCodeFileAbsolutePath()
+        // write promo code so state can access it
+    fs.writeFileSync(promoCodeOutputPath, promoCode)
```

Comment:
since this is one time only this is ok but in general we shouldn't ever write out in a sync way. 
The comment above it should have whitespace fixed though please.

Reply:
I definitely wanted to use async here but would have to convert windowsInit to be more async - at the moment it does everything sync. Happy to convert if that's better.

Reply:
fixed whitespace issue